open_prime_rando/echoes/specific_area_patches.py

```diff
@@ -95,3 +96,33 @@ def torvus_temple_crash(editor: PatcherEditor):
     # for obj in ["GibFlash", "Sound - Swamp Crate Gib"] + debris:
     #     default.remove_instance(obj)
 
+
+def agon_wastes_portal_terminal_puzzle_patch(editor: PatcherEditor):
+    """
+    Patches Agon Wastes - Portal Terminal to behave like in GC NTSC-U/PAL
+    In GC NTSC-J version a counter was added to check for each cork to be broken
+    """
+    area = editor.get_mrea(PORTAL_TERMINAL_MREA)
+
+    """
+    Remove counter increment on the 2 first corks to destroy
+    """
+    relay_ids = [0x12033A, 0x120343]  # 0x120307 is the last cork to destroy
+    for relay_id in relay_ids:
+        relay = area.get_instance(relay_id)
+
+        properties = relay.get_properties()
+        assert isinstance(properties, Relay)
+        relay.remove_connections([0x12044E])
+        relay.set_properties(properties)
+
+    """
+    Set the destroyed cork counter to expect only one cork to be destroyed
+    """
+    counter = area.get_instance(0x12044E)
+
+    assert isinstance(counter.get_properties(), Counter)
```

Comment:
The asset is redundant, as `get_properties_as` does the check for you

Reply:
I should have checked that it did. So just removing line 124 is enough?

Reply:
Yeah

Reply:
Should be good with the new commit